File: src/main/java/com/refinedmods/refinedstorage/screen/FluidStorageBlockScreen.java
package com.refinedmods.refinedstorage.screen;

import com.refinedmods.refinedstorage.RS;
import com.refinedmods.refinedstorage.blockentity.FluidStorageBlockEntity;
import com.refinedmods.refinedstorage.blockentity.NetworkNodeBlockEntity;
import com.refinedmods.refinedstorage.container.FluidStorageContainerMenu;
import net.minecraft.network.chat.Component;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.entity.player.Inventory;

public class FluidStorageBlockScreen extends StorageScreen<FluidStorageContainerMenu> {
    public FluidStorageBlockScreen(FluidStorageContainerMenu containerMenu, Inventory inventory, Component title) {
        super(
            containerMenu,
            inventory,
            title,
            new ResourceLocation(RS.ID, "textures/gui/storage.png"),
            new StorageScreenSynchronizationParameters(
                null,
                NetworkNodeBlockEntity.REDSTONE_MODE,
                FluidStorageBlockEntity.COMPARE,
                FluidStorageBlockEntity.WHITELIST_BLACKLIST,
                FluidStorageBlockEntity.PRIORITY,
                FluidStorageBlockEntity.ACCESS_TYPE
            ),
            FluidStorageBlockEntity.STORED::getValue,
            () -> (long) ((FluidStorageBlockEntity) containerMenu.getBlockEntity()).getFluidStorageType().getCapacity()
        );
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/screen/FluidStorageBlockScreen.kt
package com.refinedmods.refinedstorage.screen

import com.refinedmods.refinedstorage.container.FluidStorageContainer
import com.refinedmods.refinedstorage.tile.FluidStorageTile
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.util.text.Text
import java.util.function.Supplier

class FluidStorageBlockScreen(container: FluidStorageContainer, inventory: PlayerInventory?, title: Text?) : StorageScreen<FluidStorageContainer?>(
        container,
        inventory,
        title,
        "gui/storage.png",
        null,
        FluidStorageTile.REDSTONE_MODE,
        FluidStorageTile.COMPARE,
        FluidStorageTile.WHITELIST_BLACKLIST,
        FluidStorageTile.PRIORITY,
        FluidStorageTile.ACCESS_TYPE,
        FluidStorageTile.STORED::value,
        Supplier { (container.tile as FluidStorageTile?)!!.fluidStorageType.getCapacity() as Long }
)